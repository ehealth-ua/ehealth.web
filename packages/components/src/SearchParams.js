import React from "react";
import { Route } from "react-router-dom";
import type { Location, HistoryAction, RouterHistory } from "react-router-dom";
import { parseSearchParams, stringifySearchParams } from "@ehealth/utils";
import isEqual from "lodash/isEqual";

const setSearchParams = (
  params: {},
  method: HistoryAction,
  location: Location,
  history: RouterHistory
): void => {
  const search = parseSearchParams(location.search);
  console.log("â€‹search", search);

  if (isEqual(params, search)) return null;
  history[method]({
    ...location,
    search: stringifySearchParams(params)
  });
};

type SearchParamsProps = {
  children: (data: {
    searchParams?: { value?: string },
    setSearchParamsImmediate?: () => mixed
  }) => React.Node
};

const SearchParams = ({ children, render = children }: SearchParamsProps) => (
  <Route
    render={({ location, history }) => {
      return render({
        searchParams: parseSearchParams(location.search),
        setSearchParamsImmediate: (params, method = "replace") =>
          setSearchParams(params, method, location, history)
      });
    }}
  />
);

export default SearchParams;
