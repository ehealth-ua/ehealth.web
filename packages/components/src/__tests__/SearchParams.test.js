import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import SearchParams from "../SearchParams";

describe("SearchParams", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <MemoryRouter>
        <SearchParams>{() => <div />}</SearchParams>
      </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should return object in render prop", () => {
    const render = jest.fn(() => <div />);
    const component = renderer.create(
      <MemoryRouter>
        <SearchParams>{render}</SearchParams>
      </MemoryRouter>
    );
    expect(render).toHaveBeenCalledTimes(1);
    expect(render).toMatchSnapshot();
  });
});
