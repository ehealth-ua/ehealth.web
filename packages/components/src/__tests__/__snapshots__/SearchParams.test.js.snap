// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchParams should render correctly 1`] = `<div />`;

exports[`SearchParams should return object in render prop 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "searchParams": Object {},
        "setSearchParamsImmediate": [Function],
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": <div />,
    },
  ],
}
`;
